### PR TITLE
Install vendor tools before using them

### LINKED ISSUE #166: Install vendor tools before using them
Mimic https://github.com/nmstate/kubernetes-nmstate/pull/104 and make our development commands faster.

### PR DESCRIPTION
**Pull request by Atul9**
closes #166

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: .travis.yml [hunk 1]
```diff
@@ -11,12 +11,12 @@ script:
 
 deploy:
 - provider: script
-  script: docker login -u="$DOCKER_USER" -p="$DOCKER_PASS" quay.io && make docker-build docker-push
+  script: GINKGO_EXTRA_ARGS="--compilers=2" make all test/unit && docker login -u="$DOCKER_USER" -p="$DOCKER_PASS" quay.io && make docker-build docker-push
```
Comment (SchSeba) [inline .travis.yml]: no need to run the tests in the deployment state they run in the `make check`

File: Makefile [hunk 2]
```diff
@@ -23,15 +23,18 @@ TARGETS = \
 
 GINKGO_EXTRA_ARGS ?=
 GINKGO_ARGS ?= --v -r --progress $(GINKGO_EXTRA_ARGS)
-GINKGO ?= go run ./vendor/github.com/onsi/ginkgo/ginkgo
+GINKGO?= build/_output/bin/ginkgo
```
Comment (phoracek) [inline Makefile]: please keep the space in:

```
GINKGO ?= ...
```

File: .travis.yml (hunk 1 above)
Comment (Atul9) [reply .travis.yml]: done. I have updated the PR.

File: Makefile (hunk 2 above)
Comment (Atul9) [reply Makefile]: done. I have updated the PR.

**Comment by phoracek** [comment]
ci test please

**Comment by phoracek** [comment]
@Atul9 fails on CI https://travis-ci.com/kubevirt/cluster-network-addons-operator/builds/116733047#L491, could you look into it?

**Comment by Atul9** [comment]
ci test please

**Comment by Atul9** [comment]
@phoracek done. I have updated the PR.

**Comment by phoracek** [comment]
ci test please

**Comment by phoracek** [comment]
Still fails on STD-CI, although it passes on Travis:

```
make[1]: Leaving directory `/home/jenkins/workspace/kubevirt_cluster-network-addons-operator_standard-check-pr/cluster-network-addons-operator'
namespace/cluster-network-addons-operator created
customresourcedefinition.apiextensions.k8s.io/networkaddonsconfigs.networkaddonsoperator.network.kubevirt.io created
+ make test/e2e/lifecycle
build/_output/bin/operator-sdk test \
	local \
	./test/e2e/lifecycle \
	--namespace cluster-network-addons-operator \
	--no-setup \
	--kubeconfig ./cluster/.kubeconfig \
	--go-test-flags "-test.v -test.timeout 2h -ginkgo.v -ginkgo.noColor"
/bin/sh: build/_output/bin/operator-sdk: No such file or directory
make: *** [test/e2e/lifecycle] Error 127
```

@qinqon didn't you have a similar issue on nmstate?

**Comment by phoracek** [comment]
@SchSeba could you please take another look?

**Comment by SchSeba** [comment]
ci test please

**Pull request merged**